pkg/kube/pipeline_cache.go

```diff
@@ -0,0 +1,96 @@
+package kube
+
+import (
+	"github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
+	"github.com/jenkins-x/jx/pkg/client/clientset/versioned"
+	"github.com/jenkins-x/jx/pkg/log"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/tools/cache"
+	"sort"
+	"time"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PipelineNamespaceCache caches the pipelines for a single namespace
+type PipelineNamespaceCache struct {
+	pipelines map[string]*v1.PipelineActivity
+}
+
+// NewPipelineCache creates a cache of pipelines for a namespace
+func NewPipelineCache(jxClient versioned.Interface, ns string) *PipelineNamespaceCache {
+	pipeline := &v1.PipelineActivity{}
+	stop := make(chan struct{})
```

Comment:
It seems that nobody is closing the `stop` channel. Maybe it should be returned to the caller which might  decide when to stop the controller/cache.